### ISSUE TITLE
Simplify immediately turns itself off

### ISSUE THREAD

**Issue by adamu** [leggett/simplify#606]
I can no longer enable simplify. It appears up for a split second, and then disables itself.

The following is a video of me pressing <kbd>⌥</kbd> <kbd>s</kbd> to enable (but not disable).

https://user-images.githubusercontent.com/498229/128997735-85643441-4d3b-41a0-8c19-f1f3270c7fb9.mov

---
Simplify v.2.4.5 - Configuration: aAX ca3 caI caL cfC cfO cfQ cfR cfU changeFont composeActions darkTheme dateGroup hideListCount hideMsgCount hideTabIcons inInbox inList invertAddons invertCompose izBgDefault listWidthLg lowDensity minimizeChat msgWidthSm navOpen newUI supExtWarn - System: Mozilla/5.0 (Macintosh Intel Mac OS X 10.15 rv:90.0) Gecko/20100101 Firefox/90.0 - Window: 1680 x 1027 (100pct zoom) - Language: en-GB

**Comment by adamu** [comment]
Enabling the debug mode, I can see this in the console:

```
Checking subscription... <my workspaces email>
Could not register trial from this domain.
```

Which appears to happen when there is a `502` error from the subscription check.

**Comment by ferrreo** [comment]
I'm getting the exact same issue on firefox

**Comment by Hudyst** [comment]
I am also having this issue. Using Firefox.

**Comment by soltysh** [comment]
Same problem on my end, I checked this problem and it works fine with Chrome, so that `502 Bad Gateway` as noticed by previous commenter is not a problem with networking but seems a problem with Firefox itself. I've also tried a clean profile the same problem happens there as well.

**Comment by leggett** [comment]
Sorry everyone. Reverted the change. Shoukd work now. I'll explain when I'm fully awake.

**Comment by adamu** [comment]
Confirmed it's working again, thanks.

**Comment by leggett** [comment]
I found some evidence that someone was trying to figure out how to cheat the subscription checker. (Good luck with that dear someone.)

So I locked down the check to only work if the request header origin was mail.google.com. BUT, Firefox (and probably other browsers) proxy all requests through there servers for an extra layer of privacy / anonymity meaning the origin of the request was not mail.google.com by the time it got to me. 

I'm going to do what I should have done and just monitor any request not from mail.google.com for a couple weeks and find a better way to allow all the legit requests and block the others. 

Sorry for the downtime!
